platform/jewel/foundation/src/main/kotlin/org/jetbrains/jewel/foundation/component/LocalComponent.kt

```diff
@@ -0,0 +1,15 @@
+// Copyright 2000-2025 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.jewel.foundation.component
+
+import androidx.compose.runtime.ProvidableCompositionLocal
+import androidx.compose.runtime.staticCompositionLocalOf
+import javax.swing.JComponent
+import org.jetbrains.annotations.ApiStatus
+import org.jetbrains.jewel.foundation.ExperimentalJewelApi
+
+@Suppress("CompositionLocalAllowlist")
+@ApiStatus.Experimental
+@ExperimentalJewelApi
+public val LocalComponent: ProvidableCompositionLocal<JComponent> = staticCompositionLocalOf {
+    error("No Component provided")
+}
```

Comment:
Should we provide a function that helps with this for the stand-alone users of the library (thinking when people don't use a decorated window)?

```kotlin
fun main() {
    application {
        IntUiTheme {
            DecoratedWindow {
                Text("Foo")
            }
        }
    }
}
```

We could provide a helper function like this:

```kotlin
@Composable
@ApiStatus.Experimental
@ExperimentalJewelApi
public fun ProvideLocalComponentFromWindow(window: ComposeWindow, content: @Composable () -> Unit) {
    CompositionLocalProvider(
        LocalComponent provides window.contentPane.components.filterIsInstance<JComponent>().first(),
        content = content,
    )
}
```

Reply:
As we discussed multiple times, we need the same thing in standalone that we have in bridge (`JewelComposePanelWrapper` functions)